File: Programming Basics/C++/While loops/OldBooks/OldBooks.cpp
#include <iostream>
#include <string>
using namespace std;

int main()
{
    string bookToLookFor;
    getline(cin, bookToLookFor);

    int booksCount = 0;
    bool isFound = false;

    string input;
    while (getline(cin, input) && input != "No More Books") {
        if (input == bookToLookFor) {
            isFound = true;
            break;
        }

        booksCount++;
    }

    if (isFound) cout << "You checked " << booksCount << " books and found it." << endl;
    else cout << "The book you search is not here!" << endl << "You checked " << booksCount << " books." << endl;
}

File: Programming Basics/JavaScript/While loops/Exercises/old-books.js
function solve(input) {
    const bookToLookFor = input[0];

    let booksCount = 0;
    let isFound = false;

    let index = 1;
    while (input[index] != "No More Books") {
        if (input[index] == bookToLookFor) {
            isFound = true;
            break;
        }

        booksCount++;
        index++;
    }

    if (isFound) console.log(`You checked ${booksCount} books and found it.`);
    else {
        console.log("The book you search is not here!");
        console.log(`You checked ${booksCount} books.`);
    }
}
